File: Problem_set_1/Introductory_problems/p06_Number_Spiral/NumberSpiral.py
t = int(input())

out = []

while(t > 0):
    y, x = list(map(int, input().split()))
    z = max(x, y)

    result = 0
    if z % 2 == 0:
        if z == y:
            result = (z - 1) * (z - 1) + y
            print(result)
        else:
            result = z * z - y + 1
            print(result)
    else:
        if z == x:
            result = z * z - y + 1
            print(result)
        else:
            result = (z - 1) * (z - 1) + x
            print(result)
    t -= 1

File: Problem_set_1/Introductory_problems/p06_Number_Spiral/NumberSpiral.ts
const input = require("prompt-sync")();

let t: number = Number(input())

while(t-- > 0){
    let [y, x] = input().split(" ").map(Number)

    let z: number = Math.max(y, x)
    // let result: number;

    if(z % 2 == 0){
        if(y == z){
            console.log(z * z - x + 1);
        } else{
            console.log((z - 1) * (z - 1) + y);
        }
    } else{
        if(x == z){
            console.log(z * z - y + 1);
        } else{
            console.log((z - 1) * (z - 1) + x);
        }
    }
}